shared_model/interfaces/transaction_responses/abstract_tx_response.hpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_ABSTRACT_TX_RESPONSE_HPP
+#define IROHA_ABSTRACT_TX_RESPONSE_HPP
+
+#include "interfaces/primitive.hpp"
+#include "model/transaction_response.hpp"
+#include "utils/string_builder.hpp"
+
+namespace shared_model {
+  namespace interface {
+    /**
+     * Abstract transaction response
+     * @tparam Model - concrete model transaction response
+     */
+    template <typename Model>
+    class AbstractTxResponse
+        : public Primitive<Model, iroha::model::TransactionResponse> {
+     private:
+      /**
+       * @return string representation of class name
+       */
+      virtual std::string className() const = 0;
```

Comment:
Is it only for ```toString``` or it has a deeper meaning?

Reply:
Only for `toString`. I want to do it as a template parameter, but it is not available in an easy way.

Reply:
Why not just use rtti? `typeid(*this).name()`

Reply:
Because of no reasons to use RTTI here. `Name` may return not what you expected and the result may change on different platforms.

Reply:
Clean and consistent output better to read.